### PR TITLE
completed countries tab and added load state and error state as well

### PR DESCRIPTION
**Pull request by Sufiyan-dev**
- [x] completed Create, Read, Update operation on countries
- [x] added load state when loading data
- [x] added search filter logic and clear search handler
- [x] handle error when API fails with toast
- [x] implemented toast

### CODE REVIEW

File: src/pages/master/countries/index.tsx [hunk 1]
```diff
@@ -79,79 +74,104 @@ const Countries = () => {
...
   };
 
-  const handleSubmit = () => {
-    // let validationErrors = { name: '', age: '', city: '' };
-    // let isValid = true;
+  const handleSubmit = async () => {
+    let validationErrors = { name: '', code: '', currency: '', currency_name: '' };
+    let isValid = true;
+  
+    // Validation logic
```
Comment (abdulshakoor02) [inline src/pages/master/countries/index.tsx:109]: same code is repeated you can loop through the object for ..in loop

File: src/pages/master/countries/index.tsx [hunk 2]
```diff
@@ -79,79 +74,104 @@ const Countries = () => {
...
+    }
   
-    // if (!isValid) {
-    //   setErrors(validationErrors);
-    //   return;
-    // }
+    try {
+      let result;
+
+      console.log("form values", formValues, selectedRow?.id);
```
Comment (abdulshakoor02) [inline src/pages/master/countries/index.tsx:135]: remove console.log

**Review by abdulshakoor02 — APPROVED** [review]

File: src/pages/master/countries/index.tsx (hunk 1 above)
Comment (Sufiyan-dev) [reply src/pages/master/countries/index.tsx:109]: we can use other validation packages as well, for write a schema for validations, what specifically is been repeated the `validationErrors`. object ?

**Comment by Sufiyan-dev** [comment]
Added reusable textfield

**Comment by Sufiyan-dev** [comment]
@abdulshakoor02 we can use yup for schema validation 

```node
// Define the Yup schema
const countrySchema = yup.object().shape({
  name: yup.string().required("Name is required"),
  code: yup.string().required("Code is required"),
  currency: yup.string().required("Currency is required"),
  currency_name: yup.string().required("Currency name is required")
});
```

it will be tricky to add error helper text after validation

**Pull request merged**